internal/groups/api/requests.go

```diff
@@ -0,0 +1,126 @@
+package groups
+
+import (
+	"regexp"
+
+	"github.com/mainflux/mainflux/internal/groups"
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+var groupRegexp = regexp.MustCompile("^[A-Za-z0-9]+[A-Za-z0-9_-]*$")
+
+type createGroupReq struct {
+	token       string
+	Name        string                 `json:"name,omitempty"`
+	ParentID    string                 `json:"parent_id,omitempty"`
+	Description string                 `json:"description,omitempty"`
+	Metadata    map[string]interface{} `json:"metadata,omitempty"`
+}
+
+func (req createGroupReq) validate() error {
+	if req.token == "" {
+		return groups.ErrUnauthorizedAccess
+	}
+	if len(req.Name) > maxNameSize || req.Name == "" || !groupRegexp.MatchString(req.Name) {
+		return errors.Wrap(groups.ErrMalformedEntity, groups.ErrBadGroupName)
+	}
+
+	return nil
+}
+
+type updateGroupReq struct {
+	token       string
+	ID          string                 `json:"id,omitempty"`
```

Comment:
This field don't need to be exported. It comes as path parameter not from the body of the request.